platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java

```diff
@@ -0,0 +1,1462 @@
+   */
+  public static Expression zoom() {
+    return new Expression("zoom");
+  }
+
+  //
+  // Ramps, scales, curves
+  //
+
+  /**
+   * Produces discrete, stepped results by evaluating a piecewise-constant function defined by pairs of
+   * input and output values (\"stops\"). The `input` may be any numeric expression (e.g., `[\"get\", \"population\"]`).
+   * Stop inputs must be numeric literals in strictly ascending order.
+   * Returns the output value of the stop just less than the input,
+   * or the first input if the input is less than the first stop.
+   *
+   * @param number the input value
+   * @param stops  pair of input and output values
+   * @return expression
+   */
+  public static Expression step(Number number, Object... stops) {
+    return new Expression("step", join(new Object[] {number}, stops));
+  }
+
+  /**
+   * Produces discrete, stepped results by evaluating a piecewise-constant function defined by pairs of
+   * input and output values (\"stops\"). The `input` may be any numeric expression (e.g., `[\"get\", \"population\"]`).
+   * Stop inputs must be numeric literals in strictly ascending order.
+   * Returns the output value of the stop just less than the input,
+   * or the first input if the input is less than the first stop.
+   *
+   * @param expression the input expression
+   * @param stops      pair of input and output values
+   * @return expression
+   */
+  public static Expression step(Expression expression, Object... stops) {
+    return new Expression("step", join(new Object[] {expression}, stops));
+  }
+
+  /**
+   * Produces continuous, smooth results by interpolating between pairs of input and output values (\"stops\").
+   * The `input` may be any numeric expression (e.g., `[\"get\", \"population\"]`).
+   * Stop inputs must be numeric literals in strictly ascending order.
+   * The output type must be `number`, `array&lt;number&gt;`, or `color`.
+   *
+   * @param interpolation type of interpolation
+   * @param number        the input number
+   * @param stops         pair of input and output values
+   * @return expression
+   */
+  public static Expression interpolate(Expression interpolation, Number number, Object... stops) {
+    return new Expression("interpolate", join(new Object[] {interpolation}, join(new Object[] {number}, stops)));
+  }
+
+  /**
+   * Produces continuous, smooth results by interpolating between pairs of input and output values (\"stops\").
+   * The `input` may be any numeric expression (e.g., `[\"get\", \"population\"]`).
+   * Stop inputs must be numeric literals in strictly ascending order.
+   * The output type must be `number`, `array&lt;number&gt;`, or `color`.
+   *
+   * @param interpolation type of interpolation
+   * @param number        the input expression
+   * @param stops         pair of input and output values
+   * @return expression
+   */
+  public static Expression interpolate(Expression interpolation, Expression number, Object... stops) {
```

Comment:
I could see the case for creating a separate `Interpolation` type for the first argument here.  (So `linear()`, `exponential()`, and `cubicBezier()` would return an `Interpolation` value.)  Not sure whether it's actually better to do so, just mentioning it as an alternative.

Reply:
One benefit of doing it that way is that users couldn't do something weird like `sum(linear(), 2)`.  As it stands now, if they did this, they'd get an error message that `linear` is an unknown expression operator -- which might be confusing to them, since the `linear()` method claims to produce an expression.